processing/src/main/java/io/druid/segment/incremental/OnheapIncrementalIndex.java

```diff
@@ -237,10 +273,18 @@ protected void concurrentRemove(int offset)
   @Override
   public boolean canAppendRow()
   {
-    final boolean canAdd = size() < maxRowCount;
-    if (!canAdd) {
+    final boolean countCheck = size() < maxRowCount;
+    boolean sizeCheck = true;
+    if (maxBytesInMemory != -1) {
```

Comment:
Suggest doing `> 0` rather than `!= -1`. That way, any negative number (or zero) means unlimited.

Reply:
done